surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,360 @@
+
+### Do you often use a debugger with other programming languages (anything but Rust)?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> If a user regularly uses a debugger for other languages but not for Rust then
+> it would be interesting to know that.
+
+### Which programming language(s) do you use a debugger with?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- Go
+- C#
+
+> *Justification:*
+>
+> Debugging experiences vary widely by language - debugging a interpreted
+> language like Python is very different from a compiled language like C/C++ -
+> this will inform user expectations with Rust debuggers.
+
+### How does debugging in language X compare to Rust?
+Type: freeform text
+
+> *Justification:*
+>
+> What can Rust's debugging experience learn from other languages.
+
+### Do you often use a debugger with Rust?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> Knowing how often a debugger is used with Rust is valuable, but need to know
+> whether that is because of the quality of the experience or because it
+> is/isn't often needed.
+
+### How do you start a debugger (when debugging Rust)?
+Type: select all that apply
+
+- Core dumps
+- Attaching to a process
+- Launching executable
+- From an IDE
+
+> *Justification:*
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
+
+### What debuggers have you heard of?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> If users aren't aware of certain types of debugger than raising awareness or
+> making it easier to start these debuggers might be valuable.
+
+### What debuggers do you use with Rust?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> Debuggers that are actually used is useful.
+
+### Which debugger interfaces do you use with Rust?
+Type: select all that apply
+
+- Terminal interface
+- Visual Studio Code for `gdb`
+- Visual Studio Code for `lldb`
+- `gdb`'s TUI mode
+- Vim/Neovim's `termdebug` for `gdb`
+- Pernosco for `rr`
+- Visual Studio
+- XCode
+
+> *Justification:*
+>
+> Interface used for a debugger materially changes the experience of using a
+> debugger and the advantages/disadvantages of it.
+
+### What pain points have you experienced using a debugger with Rust?
+Type: select all that apply
+
+- Often step too many times and need to restart from scratch
+- Don't have a source view of the current line being run
+- Unintuitive interface/errors
+- Expression parser doesn't support expressions that you want to write
+- Optimized-out code
+- Conditional breakpoints don't trigger
+- Difficult to trace dataflow
+- Too much detail/irrelevant information (i.e. assembly views)
```

Comment:
This might be related, but to me "not visualizing values as expected" is a bit different. For example, seeing a `Vec<T>` displayed as a pointer, len, and capacity, instead of being able to view the elements in the vec.

Reply:
Added!